pkg/cache/cache.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2017 Envoyproxy Authors
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+// Package cache defines a configuration cache for the server
+package cache
+
+import "github.com/envoyproxy/go-control-plane/api"
+
+// ConfigCache ...
+type ConfigCache interface {
```

Comment:
Nit: I think this interface should be renamed and moved out of the package `cache`. The interface itself is not a cache, it abstracts over retrieving config. An implementation of the interface may be based on a cache, but that's not really required by the interface itself.

Reply:
What about

```go

package config

type Listener interface { Listen(...) }
```

then `//pkg/cache` contains an implementation of the `config.Listener` interface.